dependencies/README.md

```diff
@@ -0,0 +1,67 @@
+Static Linkage Checker
+======================
+
+Static Linkage Checker is a tool that finds [static linkage errors](
+../library-best-practices/glossary.md) on a classpath and reports the errors to the console.
+It scans the class files in a classpath provided as input for references
+to other classes.
+It then verifies each of the references to find linkage errors in the classpath.
+The tool also provides analysis on reachability to the errors from entry point
+classes through class usage graphs.
+
+### Use Cases
+
+There are two use cases for Static Linkage Checker:
+
+-  **For organizations** that provide multiple libraries developed by different teams,
+  the tool helps to ensure that there are no static linkage errors among the libraries and their
+  dependencies.
+
+- **For library/application developers** the tool finds static linkage
+  errors in their projects, and will help to avoid incompatible versions of libraries
+  in their dependencies.
+
+### Approach
+
+1. Given an input, the tool prepares a _linkage classpath_, from which the tool inspects Java
+  classes.
+  Note that the linkage classpath is different from the runtime classpath of the tool itself.
+
+2. The tool extracts all _references_ from the all class files in the classpath.
+
+3. The tool checks the linkage compatibility of the references through the classpath, and records
+  static linkage errors.
+  
+4. The tool traverses a _class usage graph_ to annotate the linkage errors with _reachability_.
+
+5. At the end, the tool outputs a report on the linkage errors.
+
+### Output
+
+The tool reports static linkage errors for the input linkage classpath, annotated
+with _reachability_. Each of the static linkage errors contains the information on the
+source class and the destination class of the reference, and has one of the three types:
+
+  - _Missing class type_: a class reference causes a static missing class error.
+
+  - _Missing method type_: a method reference has a static linkage conflict.
+
+  - _Missing field type_: a field reference has a static linkage conflict.
+     
+### Class Usage Graph and Reachability
+
+In order to provide a diagnosis on the output report, the tool builds _class usage graphs_,
+and annotates linkage errors with _reachability_ from _entry point classes_.
+The tool allows users to choose the scope of entry point classes:
+
+  - **Classes in the target project**: when the scope of the entry point is only the classes in the
```

Comment:
Is the input to this tool a project definition like a pom.xml file? I think it should not be. I think instead this tool should operate on pure class paths. That way we can write a Maven plugin that uses the tool, or a Gradle one, or a Bazel one.

Reply:
Added following in approach:

> The tool takes a classpath as required input, through either a list of jar and
  class files in filesystems, a list of Maven coordinates, or a Maven BOM.
  A Maven BOM is converted to a list of Maven coordinates, and a list of Maven
  coordinates is resolved to a list of jar files that consists of the artifacts
  and their dependencies.